### PR TITLE
Add function to conceal rtsp URLs in track labels

### PR DESCRIPTION
**Pull request by Bhlowe**
## Description

This change removes potentially sensitive rtsp URLs from being visible by the viewer. Currently labels are used to describe audio and video tracks. These can be stream names, device names, or rtsp urls. 

Hiding the full rtsp url is done by hashing the entire string. 

## Motivation and Context

rtsp urls can contain sensitive password information, in the form of:
rtsp://user:pass@domain/path

## Types of changes

One function was added to the PeerConnectionManager.h and PeerConnectionManager.cpp and that function is used when a label is created. 

The include <functional> was added to access the std::hash function.

### CODE REVIEW

File: src/PeerConnectionManager.cpp [hunk 1]
```diff
@@ -833,14 +833,29 @@ rtc::scoped_refptr<webrtc::AudioTrackInterface> PeerConnectionManager::CreateAud
 	if (!audioSource) {
 		RTC_LOG(LS_ERROR) << "Cannot create capturer audio:" << audiourl;
 	} else {
-		std::string label = audiourl + "_audio";
-		label.erase(std::remove_if(label.begin(), label.end(), ignoreInLabel), label.end());
-		audio_track = peer_connection_factory_->CreateAudioTrack(label, audioSource);
```
Comment (mpromonet) [inline src/PeerConnectionManager.cpp]: Do you like to remove audio_track creation ?

Comment (Bhlowe) [reply src/PeerConnectionManager.cpp]: Oh, shoot.. Need to add that missing line back:

		audio_track = peer_connection_factory_->CreateAudioTrack(label, audioSource);
Sorry!

**Comment by Bhlowe** [comment]
Hi Michael,

My need is similar to your demo page. But I don't want anyone to see the URLs/passwords, I want to add/delete streams, kick users off, not allow others to kick viewers off, not allow people to stream their own video.. Basically an admin can add/delete rtsp feeds, a viewer can view streams he is allowed to see. So basic security as well, which I was planning to do by keeping a list of authorized peer ID's that can be used with a given stream. 

why hashing only rtsp url ? I like the idea of human readable labels. The SDP stuff is already complicated enough. I only care about rtsp feeds, but if there other "sensitive" video sources, those could be concealed, or conceal everything. But it would be nice if there was a video_XXX and audio_XXX part of the label, and maybe not such a long hash! 

if rtsp url is inside a stream (either using -n/-u or config.json) the stream name is used in label ?
Yes, that is great. However, the easiest way to modify your code is to simply override the call request and inject my own options.  I believe the most flexible way to re-write the code is to get rid of the audio and video lists in peerconnection mgr and replace it with a flexible object that contains the video and audio links plus whatever other data is important (rtsp options!) 


if rtsp url with credentials is given by the viewer, the sensitive data are already available from the viewer ?  
Yes.. But imagine with a slight modification, I could go to your server, paste in the rtsp link with password, and be redirected to a new page that shows the live video. From there, I could send that http url to many others and let them see the stream without revealing the credentials. 



Background: About 12 months ago I made big changes to peerconnection and httprequests.. but the rtsp feeds got a little choppy and packets were lost and I wasn't able to debug the deep webrtc c++ code and I gave up... 12 months later I tried to merge and all the changes I made and you made created a mess. So my new plan is to make tiny changes to your code and if they make sense, I hope you'll use them.

I will fix that PR to include that missing audio, but that was a mistake.  I didn't mean to break things!

**Comment by mpromonet** [comment]
Hi,
Clearly merging after 12 months is really difficult, the WebRTC SDK is changing a lots, an interfaces also.
Last removing VideoCapturer had an important impact. 
As this is experiment I follow WebRTC latest change and do not try any backward compatibility.
In order to extend, I guess you can have your proper http interface and use the peerconnectionmanager, maybe ?
I will merge your contribution and maybe I will adopt a different way to named the stream and media.
Thanks,
Michel.

**Pull request merged**